src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -87,22 +85,22 @@ GcsServer::GcsServer(const ray::gcs::GcsServerConfig &config,
 
   auto on_done = [this](const ray::Status &status) {
     RAY_CHECK(status.ok()) << "Failed to put internal config";
-    this->main_service_.stop();
+    this->io_context_provider_.GetDefaultIoContext().stop();
   };
 
   ray::rpc::StoredConfig stored_config;
   stored_config.set_config(config_.raylet_config_list);
   RAY_CHECK_OK(gcs_table_storage_->InternalConfigTable().Put(
       ray::UniqueID::Nil(), stored_config, on_done));
   // Here we need to make sure the Put of internal config is happening in sync
-  // way. But since the storage API is async, we need to run the main_service_
+  // way. But since the storage API is async, we need to run the default io context
   // to block current thread.
   // This will run async operations from InternalConfigTable().Put() above
   // inline.
-  main_service_.run();
+  io_context_provider_.GetDefaultIoContext().run();
```

Comment:
We know the default io_context is `main_service_` since it's not part of the policy and we pass it in. Why don't use just do `main_service_.run()`?

Another way to ask is that do we want to add default io_context concept into IOContextProvider?

Reply:
Becuase later we may want to assign the default io context to a class in a policy. And since it must have the default io context, I think it'd be consistent to let it manager all io contexts incl. the default one, so I removed the field `main_service_` in GcsServer.

Reply:
> Becuase later we may want to assign the default io context to a class in a policy.

You mean different policy have different default io context?

Reply:
I mean for a same class, one policy may assign it to a dedicated io context while another policy assigns it to the default io context.

Reply:
I'm not too strong about it, just felt it adds an unnecessary layer of indirection.